uaclient/cli.py

```diff
@@ -1009,6 +1037,79 @@ def action_attach(args, *, cfg):
     )
 
 
+def _write_command_output_to_file(
+    cmd, filename: str, return_codes: List[int] = None
+) -> None:
+    """Helper which runs a command and writes output or error to filename."""
+    try:
+        out, _ = util.subp(cmd.split(), rcs=return_codes)
```

Comment:
It is, our subp is passing it directly to `Popen` and it should be a list afaics

Reply:
You're right, I realized that I've gotten away with just passing a string to Popen in the past, but it turns out that only works with `shell=True` (and even then only because of an implementation detail really). Also our `bytes_args` line would totally mess up a string thats passed int